Runtime/CesiumMetadata.cs

```diff
@@ -9,29 +10,60 @@ namespace CesiumForUnity
     [ReinteropNativeImplementation("CesiumForUnityNative::CesiumMetadataImpl", "CesiumMetadataImpl.h")]
     public partial class CesiumMetadata : MonoBehaviour
     {
-        internal partial void loadMetadata(Transform transform, int triangleIndex, MetadataProperty[] properties);
+         /// <summary>
+         /// Contains information to identify a feature within a glTF.
+         /// </summary>
+         internal class FeatureReference {
+            public string className {get; set;}
+            public long featureID {get; set;}
+            public string featureTable {get; set;}
+            public int numProperties {get; set;}
+        }
+
+        public class Feature {
+            public string className;
+            public string featureTableName;
+            public Dictionary<string, MetadataProperty> properties;
+        }
 
-        private partial int getNumberOfProperties(Transform transform);
+        internal partial int getNumberOfFeatures(Transform transform);
+        internal partial void getFeatureReferences(Transform transform, int triangleIndex, FeatureReference[] references);
+        internal partial void getProperties(Transform transform, FeatureReference reference, MetadataProperty[] properties);
 
         /// <summary>
-        /// Gets the properties corresponding to a particular triangle in a tile.
+        /// Gets the features corresponding to a particular triangle in a tile.
         /// </summary>
         /// <param name="transform">The <code>Transform</code> of the tile from which to obtain properties.</param>
         /// <param name="triangleIndex">The index of the tile's triangle for which to obtain properties.</param>
-        /// <returns>An array of properties associated with the triangle.</returns>
+        /// <returns>An array of features associated with the triangle.</returns>
         /// <remarks>
         /// The information to pass to this function can be obtained using the <code>Physics.Raycast</code>
         /// function.
         /// </remarks>
-        public MetadataProperty[] GetProperties(Transform transform, int triangleIndex)
-        {
-            int numberOfProperties = getNumberOfProperties(transform);
-            MetadataProperty[] properties = new MetadataProperty[numberOfProperties];
-            for(int i = 0; i < numberOfProperties; i++){
-                properties[i] = new MetadataProperty();
+        public Feature[] GetFeatures(Transform transform, int triangleIndex){
```

Comment:
Did you consider implementing this entire function in C++? I guess Reinterop might struggle with the Dictionary?

Reply:
I'm getting stuck at how to create the new objects, new MetadataProperty[] and new Feature[] can only be called in C#.